internal/handlers/pg/pgdb/query.go

```diff
@@ -106,6 +109,35 @@ func (pgPool *Pool) QueryDocuments(ctx context.Context, tx pgx.Tx, sp *SQLParam)
 	return fetchedChan, nil
 }
 
+// GetDocuments returns an iterator to fetch documents for given SQLParams.
+// If the collection doesn't exist, it returns nil and no error.
+// If an error occurs, it returns nil and that error, possibly wrapped.
+func (pgPool *Pool) GetDocuments(ctx context.Context, tx pgx.Tx, sp *SQLParam) (*Iterator, error) {
+	q, args, err := buildQuery(ctx, tx, sp)
+	if err != nil {
+		if errors.Is(err, ErrTableNotExist) {
+			return nil, nil
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+
+	rows, err := tx.Query(ctx, q, args...)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	runtime.SetFinalizer(rows, func(rows pgx.Rows) {
+		rows.Close()
+		pgPool.Config().ConnConfig.Logger.Log(
```

Comment:
@AlekSi finalizer check (to make it visible in tests and in logs in general)

Reply:
@AlekSi but iterator is not responsible for `rows` anymore (as you told me to do). Imagine, we had a channel instead of `rows`, and we close this channel in the finalizer. I can add a comment to the `newIterator` that whoever passes rows must close them.

Reply:
fixed (tested)